page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/data/DataParser.java

```diff
@@ -59,15 +40,23 @@ public void replaceStepPlaceholders(List<CucumberFeature> cucumberFeatures) thro
 
                 for (Step step : steps) {
                     FieldUtils.writeField(step, "argument", replaceArgumentPlaceholders(step.getArgument()), true);
-                    FieldUtils.writeField(step, "text", replaceStepPlaceholders(step.getText()), true);
+                    FieldUtils.writeField(step, "text", replaceDataPlaceholders(step.getText()), true);
                 }
             }
         }
     }
 
+    private List<Tag> getScenarioTags(ScenarioDefinition scenarioDefinition) {
+        try {
+            return (List<Tag>) FieldUtils.readField(scenarioDefinition, "tags", true);
+        } catch (IllegalArgumentException | IllegalAccessException e) {
+            return Collections.EMPTY_LIST;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace "Collections.EMPTY_LIST" by "Collections.emptyList()". ([squid:S1596](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS1596))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=89&issues=AWXBovs6lgbYnohYjhZS&open=AWXBovs6lgbYnohYjhZS)

[ISSUE_KEY]: AWXBovs6lgbYnohYjhZS